### ISSUE TITLE
Consider narrowing the strings allowed for condition records in ConditionExpressionJSON 

### ISSUE THREAD

**Issue by derekpierre** [nucypher/nucypher-ts#230]
See conversation from PR review here - https://github.com/nucypher/nucypher-ts/pull/229#discussion_r1235352880.

Maybe we can use the schema fields defined for underlying conditions to limit the record keys that can be used with `ConditionExpressionJSON` object.

**Comment by piotr-roslaniec** [comment]
Someone recommended using Zod, which supports static typing of object schemas: https://github.com/nucypher/nucypher-ts/issues/94

It would solve the ambiguity of keys here, and the ambiguity of types in a bunch of other places.

**Comment by derekpierre** [comment]
Could be interesting. We probably would need to push out any possible changes to Zod to a post-7.0.0 release. Also, it remains to be seen what ripple effects moving condition code to `nucypher-core` post 7.0.0 would have in this area. Not sure the level of work, but adding a whole new library at this later stage for v7.0.0 may not be worth it.

Generally, I think this specification:
```typescript
type ConditionProps = 'operator' | 'method' | 'contractAddress';

export type ConditionExpressionJSON = {
  version: string;
  condition: Record<ConditionProps, unknown>;
};
```

should not try to specify `ConditionProps` since it is too low level, but rather should be something like this (and I don't know if this is allowed or not but):
```typescript
type ConditionRecord = TimeConditionRecord | RPCConditionRecord | ContractConditionRecord | ...

export type ConditionExpressionJSON = {
  version: string;
  condition: ConditionRecord
};
```

That way you can be a lot more specific, and reuse the schema information from the various conditions themselves, instead of trying to define something separate.

In any case, this is not meant to be a comment on Zod, but rather a comment on what that expression should really be doing.

**Comment by piotr-roslaniec** [comment]
Because of type ambiguity, the resulting `ConditionRecord` simplifies to `Record<string, Joi.schema>`
```
type ConditionRecord = TimeConditionRecord | RPCConditionRecord | ContractConditionRecord
```

I agree that issue could be reviewed post-7.0.0 and after reviewing possible changes to `nucypher-core`. I just wanted to highlight the subject of object schema typing.

**Comment by derekpierre** [comment]
Closed via #267.
